src/redux/explorer.js

```diff
@@ -265,7 +266,11 @@ export const explorerInit = () => async (dispatch, getState) => {
 
   // Fallback to the testnet data provider
   // if we're not on mainnnet
-  if (network !== NetworkTypes.mainnet || forceFallbackProvider) {
+  const provider = await getProviderForNetwork(network);
+  const providerUrl = provider?.connection?.url;
+  const isHardHat =
+    providerUrl?.startsWith('http://') && providerUrl?.endsWith('8545');
```

Comment:
do we really need this last portion `providerUrl?.endsWith('8545')` ?

Reply:
I'm not sure I have enough context on all the possible providerUrls to answer this question. Happy to remove it if we're certain it's unnecessary.

Reply:
that's hardhat's default port. I'd like to keep it but maybe move it to a constant somewhere else